crates/sui-types/src/zk_login_authenticator.rs

```diff
@@ -164,19 +180,40 @@ impl AuthenticatorTrait for ZkLoginAuthenticator {
             SignatureScheme::ZkLoginAuthenticator,
         )?;
 
-        // Use flag || pk_bytes.
-        let mut extended_pk_bytes = vec![self.user_signature.scheme().flag()];
-        extended_pk_bytes.extend(self.user_signature.public_key_bytes());
-        verify_zk_login(
-            &self.inputs,
-            self.max_epoch,
-            &extended_pk_bytes,
-            &aux_verify_data.oidc_provider_jwks,
-            &aux_verify_data.zk_login_env,
-        )
-        .map_err(|e| SuiError::InvalidSignature {
-            error: e.to_string(),
-        })
+        if zklogin_inputs_cache.is_some()
+            && zklogin_inputs_cache
+                .clone()
+                .unwrap()
+                .is_cached(&self.hash_inputs())
+        {
+            // If the zklogin inputs hits the cache, we don't need to verify the zklogin again that contains the heavy computation.
+            Ok(())
+        } else {
+            // if it is not cached, we verify the full zklogin inputs.
+            // build extended_pk_bytes as flag || pk_bytes.
+            let mut extended_pk_bytes = vec![self.user_signature.scheme().flag()];
+            extended_pk_bytes.extend(self.user_signature.public_key_bytes());
+            let res = verify_zk_login(
+                &self.inputs,
+                self.max_epoch,
+                &extended_pk_bytes,
+                &aux_verify_data.oidc_provider_jwks,
+                &aux_verify_data.zk_login_env,
+            )
```

Comment:
the cache key is defined as bcs bytes for self.inputs, self.max_epoch, extended_pk_bytes. 

&aux_verify_data.oidc_provider_jwks not included bc the cache refreshed per epoch, if jwks changes, its implied that epoch cache is refreshed and the cache should also misses and verify_zklogin_inputs 
&aux_verify_data.zk_login_env: not included bc sui that runs in devnet vs testnet/mainnet will not change.